commit 07c25904c3aa504d221ea2f673db8bf1271eecce
Author: Lukáš Tinkl <lukas@kde.org>
Date:   Fri Jun 14 21:56:24 2013 +0200

    Up/Down keys should not move to the left view when on the first/last position
    
    Tested the whole kickoff menu keyboard navigation, everything works as
    expected now: Right key into the submenu, left key to the parent menu;
    when in the toplevel, it moves to the view on the right/left
    when pressing Right/Left arrow keys respectively
    
    BUG: 310166

diff --git a/plasma/desktop/applets/kickoff/ui/flipscrollview.cpp b/plasma/desktop/applets/kickoff/ui/flipscrollview.cpp
index 1d48f5b..78af286 100644
--- a/plasma/desktop/applets/kickoff/ui/flipscrollview.cpp
+++ b/plasma/desktop/applets/kickoff/ui/flipscrollview.cpp
@@ -309,9 +309,6 @@ QModelIndex FlipScrollView::moveCursor(CursorAction cursorAction, Qt::KeyboardMo
         } else if (currentIndex().row() > 0) {
             index = currentIndex().sibling(currentIndex().row() - 1,
                                            currentIndex().column());
-        } else if (currentIndex().row() == 0) {
-            kDebug() << "we are in row 0, processing Key_Up";
-            emit focusNextViewLeft();
         }
         break;
     case MoveDown:
@@ -321,9 +318,6 @@ QModelIndex FlipScrollView::moveCursor(CursorAction cursorAction, Qt::KeyboardMo
                    model()->rowCount(currentIndex().parent()) - 1) {
             index = currentIndex().sibling(currentIndex().row() + 1,
                                            currentIndex().column());
-        } else {
-            kDebug() << "we are in Last row, processing Key_Down";
-            emit focusNextViewLeft();
         }
         break;
     case MoveLeft:
@@ -534,8 +528,6 @@ void FlipScrollView::paintEvent(QPaintEvent * event)
         }
     }
 
-    QRectF eventRect = event->rect();
-
     // draw navigation
     QStyle::State state = 0;
     if (currentRoot.isValid()) {
